### PR TITLE
Add Grafana data source with results caching disabled in microservices Docker Compose stack

### PR DESCRIPTION
**Pull request by charleskorn**
#### What this PR does

This PR adds a second Grafana data source with results caching disabled to the microservices Docker Compose stack.

This is useful when debugging incorrect query results.

#### Which issue(s) this PR fixes or relates to

(none)

#### Checklist

- [n/a] Tests updated.
- [n/a] Documentation added.
- [n/a] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [n/a] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: development/mimir-microservices-mode/config/datasources.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  access: proxy
+  uid: mimir-no-caching
+  orgID: 1
+  url: http://nginx:8080/prometheus
+  jsonData:
+    prometheusType: Mimir
+    exemplarTraceIdDestinations:
+      - name: traceID
+        datasourceUid: jaeger
+    httpHeaderName1: Cache-Control
```
Comment (jhesketh) [inline development/mimir-microservices-mode/config/datasources.yaml:27]: Does this not need a value?

**Review by jhesketh — COMMENTED** [review]
change lgtm, but I'm a bit confused about how the header value is set?

File: development/mimir-microservices-mode/config/datasources.yaml (hunk 1 above)
Comment (charleskorn) [reply development/mimir-microservices-mode/config/datasources.yaml:27]: It is set in `secureJsonData` below.

Comment (jhesketh) [reply development/mimir-microservices-mode/config/datasources.yaml:27]: ah thanks. Don't know how I missed that

**Review by jhesketh — APPROVED** [review]

**Pull request merged**